### PR TITLE
Fix empty checklist markdown

### PR DESCRIPTION
**Pull request by gonzalovelasco**

### CODE REVIEW

File: lib/screens/note/note_editor.dart [hunk 1]
```diff
@@ -380,13 +380,10 @@ class _NoteEditorState extends ConsumerState<NoteEditor> {
     noteLoading.update((_) => false);
   }
 
-  bool onChecklistPressed() {
+  void onChecklistPressed() {
     if (contentController.text.isEmpty) {
-      setState(() {
-        checkListEnabled = true;
-      });
+      widget.onCheckListEnabled!();
```
Comment (matthewwong525) [inline lib/screens/note/note_editor.dart:385]:
```suggestion
      widget.onCheckListEnabled?.call();
```

Comment (matthewwong525) [reply lib/screens/note/note_editor.dart:385]: let's avoid using the ! . it breaks things.

File: lib/screens/note/note_editor.dart [hunk 2]
```diff
@@ -380,13 +380,10 @@ class _NoteEditorState extends ConsumerState<NoteEditor> {
     noteLoading.update((_) => false);
   }
 
-  bool onChecklistPressed() {
```
Comment (matthewwong525) [inline lib/screens/note/note_editor.dart:383]: changing this breaks a change i made in the content field. see gif below. maybe solution is passing the state into the contentfield, you can decide how it should be done. also can you add a test for this as well?
Export-1693499925469

**Review by matthewwong525 — CHANGES_REQUESTED** [review]
When creating from an empty note in edit mode, it should create a new list item (I only see + LIst item button, no place to input text for checklist) but it doesn't. See below also my other comments. Apart from these, good job gonzalo. Im proud of you!
Export-1693500176198

**Comment by gonzalovelasco** [comment]
> When creating from an empty note in edit mode, it should create a new list item (I only see + LIst item button, no place to input text for checklist) but it doesn't. See below also my other comments. Apart from these, good job gonzalo. Im proud of you! Export-1693500176198

Fixed 09a58e0544bc049ec9ca08c5b9b0a46841860d04

**Review by matthewwong525 — APPROVED** [review]
LGTM, give it a look thru

File: lib/screens/note/note_editor.dart [hunk 3]
```diff
@@ -378,6 +378,11 @@ class _NoteEditorState extends ConsumerState<NoteEditor> {
     noteLoading.update((_) => false);
   }
 
+  String get replacedText {
+    return contentController.text
```
Comment (matthewwong525) [inline lib/screens/note/note_editor.dart:382]:
```suggestion
    // TODO: fix work around (https://github.com/fleetingnotes/fleeting-notes-flutter/pull/906)
    return contentController.text
```

File: lib/screens/note/note_editor.dart [hunk 4]
```diff
@@ -378,6 +378,11 @@ class _NoteEditorState extends ConsumerState<NoteEditor> {
...
+        .replaceAll(RegExp(r"- \[ \] ?(\n|$)"), "- [ ]\n");
```
Comment (matthewwong525) [inline lib/screens/note/note_editor.dart:383]: This doesn't handle the case where the checkbox is ticked `- [x]`

**Review by matthewwong525 — CHANGES_REQUESTED** [review]
almost there, good work finding this workaround!

File: lib/screens/note/note_editor.dart (hunk 4 above)
Comment (gonzalovelasco) [reply lib/screens/note/note_editor.dart]: Fixed 06d56fbb3fb9dc2036429c7a6473bd8a1b9f56cd

File: lib/screens/note/note_editor.dart (hunk 3 above)
Comment (gonzalovelasco) [reply lib/screens/note/note_editor.dart:383]: Fixed 06d56fbb3fb9dc2036429c7a6473bd8a1b9f56cd

**Review by matthewwong525 — APPROVED** [review]
LGTM

**Pull request merged**